### DISCUSSION TITLE
Comment counter issue in React

### DISCUSSION THREAD

**Discussion by xRitsu** [umputun/remark42#1593] [General]
When using dynamic routing on a single-site project with the use of React and NextJS, using a link that leads to another page, then navigating back to the page where Remark42 is displayed, causes the comment counter to disappear.
Untested with pure React, all I know is that it didn't work with the `Link` component from NextJS (and their Router API as well), so it probably doesn't work properly in general.

Any possible workaround to this? I am now forced to fetch the number of comments manually, but this does not have dynamic updates.

**Comment by paskal** [comment]
I think it wouldn't be possible to help you without seeing the site and the code if possible. Is the site's source code published, and can you share the URL?

**Reply by xRitsu** [reply]
Hello, sorry for taking that long to reply. The source code is not public, but this is the url: https://ordinary-site.eu/babel

**Comment by akellbl4** [comment]
It's not an issue connected to React or whatever, the widget were desined for sites with traditional page navigation so it doesn't listen to any non-reload navigation (which is used in NextJS in particular). A workaround for it is to reinitialize the widged on every navigation. I know it's not ideal and it's room for improvement in general.

**Reply by xRitsu** [reply]
Hi, sorry for late reply.

_It's not an issue connected to React or whatever_
I see. How can I accomplish this? Because I indeed thought about it and even tried to somehow implement it, but the widget didn't want to appear again.

**Reply by meinto** [reply]
Can you share implementation details for this workaround with us?
